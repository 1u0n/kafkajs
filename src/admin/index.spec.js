const createAdmin = require('./index')
const { createCluster, newLogger } = require('testHelpers')

describe('Admin', () => {
  it('gives access to its logger', () => {
    expect(
      createAdmin({
        cluster: createCluster(),
        logger: newLogger(),
      }).logger()
    ).toMatchSnapshot()
  })

  it('emits connection events', async () => {
    const admin = createAdmin({
      cluster: createCluster(),
      logger: newLogger(),
    })
    const connectListener = jest.fn().mockName('connect')
    const disconnectListener = jest.fn().mockName('disconnect')
    admin.on(admin.events.CONNECT, connectListener)
    admin.on(admin.events.DISCONNECT, disconnectListener)

    await admin.connect()
    expect(connectListener).toHaveBeenCalled()

    await admin.disconnect()
    expect(disconnectListener).toHaveBeenCalled()
  })
})
