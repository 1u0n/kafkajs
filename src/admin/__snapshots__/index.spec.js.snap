// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin gives access to its logger 1`] = `
Object {
  "debug": [Function],
  "error": [Function],
  "info": [Function],
  "warn": [Function],
}
`;
